packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -58,14 +62,46 @@ export async function tryHotswapDeployment(
 }
 
 async function findAllHotswappableChanges(
-  stackChanges: cfn_diff.TemplateDiff, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  stackChanges: cfn_diff.TemplateDiff,
+  evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  sdk: ISDK,
+  rootStackArtifact: cxapi.CloudFormationStackArtifact,
+  nestedStackNames: {[key: string]: NestedStackNames},
 ): Promise<HotswapOperation[] | undefined> {
   const resourceDifferences = getStackResourceDifferences(stackChanges);
 
   let foundNonHotswappableChange = false;
   const promises: Array<Array<Promise<ChangeHotswapResult>>> = [];
+  const hotswappableResources = new Array<HotswapOperation>();
+
   // gather the results of the detector functions
   for (const [logicalId, change] of Object.entries(resourceDifferences)) {
+    if (change.newValue?.Type === 'AWS::CloudFormation::Stack' && change.oldValue?.Type === 'AWS::CloudFormation::Stack') {
+      const nestedStackParameters = await evaluateCfnTemplate.evaluateCfnExpression(change.newValue?.Properties?.Parameters);
+
+      const nestedStackName = nestedStackNames[logicalId].stackName;
+      // the stack name could not be found in CFN, so this is a newly created nested stack
+      if (!nestedStackName) {
+        return undefined;
+      }
+
+      const evaluateNestedCfnTemplate = evaluateCfnTemplate.createNestedEvaluateCloudFormationTemplate(
+        new LazyListStackResources(sdk, nestedStackName), change.newValue?.Properties?.NestedTemplate, nestedStackParameters,
+      );
+
+      const nestedDiff = cfn_diff.diffTemplate(
+        change.oldValue?.Properties?.NestedTemplate, change.newValue?.Properties?.NestedTemplate,
+      );
```

Comment:
the object returned by `diff` contains only the changes made to the top level resources, and it does not contain the `diff` made to the nested resources. That is, it contains a bunch of `ResourceDifference`s, and those `ResourceDifference`s do not contain nested `diff`s, so we have to recalculate it here. We could change the way `diffTemplate()` works, and have it store nested diffs (this could let us make a nicer nested stack diff output with some extra work), but I think that belongs in a separate PR.

Reply:
#19162